pkg/testmachinery/testrun/validation.go

```diff
@@ -41,7 +42,7 @@ func Validate(tr *tmv1beta1.Testrun) error {
 	k := reflect.ValueOf(tr.Spec.Kubeconfigs)
 	typeOfK := k.Type()
 	for i := 0; i < k.NumField(); i++ {
-		if err := validateKubeconfig(fmt.Sprintf("spec.kubeconfig.%s", typeOfK.Field(i).Name), k.Field(i).String()); err != nil {
+		if err := validateKubeconfig(fmt.Sprintf("spec.strconf.%s", typeOfK.Field(i).Name), k.Field(i).Interface().(*strconf.StringOrConfig)); err != nil {
```

Comment:
What is typeOfK? Maybe rename in typeOfKind?

Reply:
K is not kind it is the type of the variable k. This is necessary as we doing reflection with unknown types.